mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java

```diff
@@ -93,7 +116,43 @@ public MesosSchedulerClient(
         return createPost.call(subscription)
             .flatMap(httpClient::submit)
             .subscribeOn(Schedulers.io())
-            .flatMap(AbstractHttpContentHolder::getContent)
+            .flatMap(resp -> {
+                final HttpResponseStatus status = resp.getStatus();
+                final int code = status.code();
+
+                if (code == 200) {
+                    return resp.getContent();
+                } else {
+                    //TODO: Figure out how to get these subscribe exceptions to propagate up so that the observable dies
```

Comment:
What happens currently? And did you want to take care of this now or file an issue?